tests/integration_tests/end_to_end_test.py

```diff
@@ -53,7 +67,8 @@ def test_end_to_end_workflow(tmp_dir):
         build_model,
         objective='val_accuracy',
         max_trials=20,
-        directory=tmp_dir)
+        distribution_strategy=distribution_strategy,
+        directory='test_dir')
```

Comment:
When this test is run on CI, the `directory` argument should be generated by pytest (so that the test is thread-safe, otherwise we'd be sharing the same directory across test runs, potentially).

Reply:
Ah, ok. Didn't understand what fixture was doing